storage/rocksdb/ha_rocksdb.cc

```diff
@@ -10560,6 +10569,8 @@ int ha_rocksdb::index_end() {
   active_index = MAX_KEY;
   in_range_check_pushed_down = FALSE;
 
+  mrr_free();
```

Comment:
Minor point: can this be conditioned on mrr_uses_default_impl? Otherwise, it seems like every index_end() call always goes through mrr_free(), even though mrr is never invoked. It might be safer to always call mrr_free(), but it seems unnecessary most of the time.

Reply:
Yes. I think it's better to condition on `mrr_rowid_reader`  (which is always present when using a non-default MRR implementation). Done.